### PR TITLE
feat: added drawer component

### PR DESCRIPTION
**Pull request by tanishq20**
Added drawer component

- Permanent drawer

Live preview link: https://deploy-preview-20--compact-ui.netlify.app/docs/components/drawer
preview image

### CODE REVIEW

File: stylesheets/components/drawer.css [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+  color: var(--black);
+  font-size: 1.6rem;
+  gap: 20px;
+  padding: 10px;
+  border-radius: 5px;
+  margin-top: 5px;
+  margin-bottom: 5px;
```
Comment (nakulsharma15) [inline stylesheets/components/drawer.css:55]: You can extract this in a single line (margin shorthand property)

File: stylesheets/components/drawer.css [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+.drawer-list-item {
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+  color: var(--black);
+  font-size: 1.6rem;
+  gap: 20px;
+  padding: 10px;
```
Comment (nakulsharma15) [inline stylesheets/components/drawer.css:52]: Can use rem here instead of px

File: stylesheets/components/drawer.css [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  overflow: auto;
+  height: 100%;
+}
+
+.permanent-drawer-right-container-header {
+  color: var(--white);
+  background-color: var(--indigo-500);
+  padding: 2rem;
+  font-size: 2.4rem;
+  font-weight: 600;
```
Comment (nakulsharma15) [inline stylesheets/components/drawer.css:94]: you can use variables for font-weight

File: stylesheets/components/drawer.css [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+.permanent-drawer-right-container-content {
+  color: var(--black-500);
+  padding: 10px;
+  font-size: 1.6rem;
+  font-weight: 600;
```
Comment (nakulsharma15) [inline stylesheets/components/drawer.css:101]: same as above

**Review by anushasunil — APPROVED** [review]
Nothing to suggest. Everything looks good. Great work 👍

File: stylesheets/components/drawer.css [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  margin: auto;
+  width: 90%;
+  font-weight: 600;
+  height: 100%;
+  overflow-y: auto;
+}
+
+.drawer-list hr {
+  margin-top: 10px;
+  margin-bottom: 10px;
```
Comment (nakulsharma15) [inline stylesheets/components/drawer.css:42]: you can use margin shorthand property here

**Review by vivekmehto — APPROVED** [review]
Looks good.

File: stylesheets/components/drawer.css [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+.drawer-header {
+  flex-shrink: 0;
+  min-height: 6.4rem;
+  padding: 10px;
+}
+
+.drawer-title {
+  color: var(--secondary-color);
+  font-weight: 700;
```
Comment (nakulsharma15) [inline stylesheets/components/drawer.css:20]: you can use variables for font-weight

File: stylesheets/components/drawer.css [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  font-size: 2rem;
+}
+
+.drawer-subtitle {
+  color: var(--blueGrey-500);
+  font-weight: 600;
```
Comment (nakulsharma15) [inline stylesheets/components/drawer.css:26]: same as above

File: stylesheets/components/drawer.css [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+  font-size: 1.6rem;
+}
+
+.drawer-list {
+  display: flex;
+  flex-direction: column;
+  margin: auto;
+  width: 90%;
+  font-weight: 600;
```
Comment (nakulsharma15) [inline stylesheets/components/drawer.css:35]: same as above

File: docs/components/drawer.html [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+                        style="width: 100%; height: 450px; border:0; transform: scale(1); overflow:hidden;"
```
Comment (nakulsharma15) [inline docs/components/drawer.html:117]: you can extract this style in a class as you're using it elsewhere too

**Review by nakulsharma15 — COMMENTED** [review]
Suggested some minor changes

**Pull request merged**